src/Setup/WixBA/RootView.xaml

```diff
@@ -1,125 +1,487 @@
+                                        AutomationProperties.HelpText="Click to try again"
+                                        AutomationProperties.Name="Try Again?"
+                                        Command="{Binding TryAgainCommand}"
+                                        Content="Try Again?"
+                                        Style="{DynamicResource ActionButtonStyle}"
+                                        Visibility="{Binding TryAgainEnabled,
+                                                             Converter={StaticResource BooleanToVisibilityConverter}}" />
+                            </StackPanel>
+
+                            <!--  Final Actions - only one of these is expected to be displayed, if any  -->
+                            <StackPanel x:Name="CloseBtnDcIsland"
+                                        HorizontalAlignment="Right"
+                                        DataContext="{Binding InstallationViewModel}"
+                                        Style="{DynamicResource ActionBtnStkPnlStyle}">
+                                <Button x:Name="CloseButton"
+                                        AutomationProperties.HelpText="Click to Close"
+                                        AutomationProperties.Name="Close"
+                                        Command="{Binding CloseCommand}"
+                                        Content="Close"
+                                        Style="{DynamicResource FinalActionButtonStyle}"
+                                        Visibility="{Binding CloseEnabled,
+                                                             Converter={StaticResource BooleanToVisibilityConverter}}" />
+                            </StackPanel>
+                            <StackPanel x:Name="CancelBtnDcIsland"
+                                        HorizontalAlignment="Right"
+                                        DataContext="{Binding}"
+                                        Style="{DynamicResource ActionBtnStkPnlStyle}">
+                                <Button AutomationProperties.HelpText="Press to Cancel"
+                                        AutomationProperties.Name="Cancel"
+                                        Command="{Binding CancelCommand}"
+                                        Content="Cancel"
+                                        Style="{DynamicResource FinalActionButtonStyle}"
+                                        Visibility="{Binding CancelEnabled,
+                                                             Converter={StaticResource BooleanToVisibilityConverter}}" />
+                            </StackPanel>
+
+                        </StackPanel>
+                    </Grid>
+                </DockPanel>
+
+                <StackPanel x:Name="FinalStatusArea"
+                            HorizontalAlignment="Stretch"
+                            DockPanel.Dock="Bottom">
```

Comment:
I agree that there are multiple ways to do layout in WPF.  DockPanel's do
not have to have a single DockPanel.Dock specification per syblings, and
the use of multiple docking is a common WPF practice that I think is
appropriate here.  I'm sure that it could be done with a Grid or some
other preferred approach depending on the reviewer.  I will ponder this,
since I mean to please.

On Sun, Jul 31, 2016 at 11:59 PM, Sean Hall <notifications@github.com>
wrote:

> In src/Setup/WixBA/RootView.xaml
> <https://github.com/wixtoolset/wix4/pull/208#discussion_r72924244>:
>
> > +                                <Button AutomationProperties.HelpText="Press to Cancel"
> > +                                        AutomationProperties.Name="Cancel"
> > +                                        Command="{Binding CancelCommand}"
> > +                                        Content="Cancel"
> > +                                        Style="{DynamicResource FinalActionButtonStyle}"
> > +                                        Visibility="{Binding CancelEnabled,
> > +                                                             Converter={StaticResource BooleanToVisibilityConverter}}" />
> > +                            </StackPanel>
> > +
> > +                        </StackPanel>
> > +                    </Grid>
> > +                </DockPanel>
> > +
> > +                <StackPanel x:Name="FinalStatusArea"
> > +                            HorizontalAlignment="Stretch"
> > +                            DockPanel.Dock="Bottom">
>
> The sibling StackPanel above is already assigned to
> DockPanel.Dock="Bottom". Do not assign multiple controls the same docking
> location. There are multiple ways to layout controls like this, my
> preference is to use Grids everywhere, I seldom use DockPanels or
> StackPanels.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/wixtoolset/wix4/pull/208/files/28721ab8e4985a5641b3efd7975a67cecd8f8d4a#r72924244>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AHyCLZ2R91vuoG3sEo8iSfYV8zWianEKks5qbX0XgaJpZM4JVm8P>
> .
>